spinnaker-monitoring-daemon/spinnaker-monitoring/spectator_metric_transformer.py

```diff
@@ -24,6 +24,17 @@
 import re
 import yaml
 
+
+# see https://stackoverflow.com/questions/1175208/\
+#     elegant-python-function-to-convert-camelcase-to-snake-case
+_SNAKE_CASE_RE = re.compile('((?<=[a-z0-9])[A-Z]|(?!^)(?<!_)[A-Z](?=[a-z]))')
```

Comment:
Might be a bit more robust to use a library such as https://github.com/jpvanhal/inflection for this; but at least this is unit tested so I suppose we know it's correct.